sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,636 @@
+        This API specifically enables the auto switching or manual switching feature on the muxcable,
+        depending upon the mode entered by the user.
+        Autoswitch feature if enabled actually does an automatic toggle of the mux in case the active
+        side link goes down and basically points the mux to the other side.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+             mode:
+                 One of the following predefined constants:
+                 SWITCHING_MODE_AUTO
+                 SWITCHING_MODE_MANUAL
+
+                 specifies which type of switching mode we set the muxcable to
+                 either SWITCHING_MODE_AUTO or SWITCHING_MODE_MANUAL
+
+        Returns:
+            a Boolean, True if the switch succeeded and False if it did not succeed.
+        """
+
+        raise NotImplementedError
+
+    def get_switching_mode(self):
+        """
+        This API specifically returns which type of switching mode the cable is set to auto/manual
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            One of the following predefined constants:
+               SWITCHING_MODE_AUTO if auto switch is enabled.
+               SWITCHING_MODE_MANUAL if manual switch is enabled.
+        """
+
+        raise NotImplementedError
+
+    def get_nic_temperature(self):
+        """
+        This API specifically returns nic temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_temperature(self):
+        """
+        This API specifically returns local ToR temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def get_nic_voltage(self):
+        """
+        This API specifically returns nic voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_voltage(self):
+        """
+        This API specifically returns local ToR voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def enable_prbs_mode(self, target, mode_value, lane_map):
```

Comment:
Everything below here appears to be debug functions, correct? I suggest adding a comment to visibly separate this section. Something like:

```python
#############################################################################################
###                                  Debug Functionality                                  ### 
#############################################################################################
```

Reply:
added